.github/workflows/tests.yml

```diff
@@ -0,0 +1,171 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  tests:
+    env:
+      COLORTERM: truecolor
+      NPROC: 2
+      TERM: xterm
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - {
+            icon: 🍎,
+            os: macos,
+            shell: bash --noprofile --norc -eo pipefail
+          }
+          - {
+            icon: 🐧,
+            os: ubuntu,
+            shell: bash --noprofile --norc -eo pipefail
+          }
+          - {
+            icon: 🏁,
+            os: windows,
+            shell: msys2
+          }
+    name: ${{ matrix.platform.icon }} ${{ matrix.platform.os }}
+    runs-on: ${{ matrix.platform.os }}-latest
+    defaults:
+      run:
+        shell: ${{ matrix.platform.shell }} {0}
+
+    steps:
+
+      - name: Install tools and libraries via APT
+        if: matrix.platform.os == 'ubuntu'
+        run: |
+          sudo apt install -y \
+            build-essential \
+            cmake \
+            doctest-dev \
+            ffmpeg \
+            libavcodec-dev \
+            libavformat-dev \
+            libavutil-dev \
+            libncurses-dev \
+            libqrcodegen-dev \
+            libreadline-dev \
+            libswscale-dev \
+            libunistring-dev \
+            pandoc \
+            pkg-config \
+            python3-cffi \
+            python3-dev \
+            python3-setuptools
+
+      - name: Install tools and libraries via Homebrew
+        if: matrix.platform.os == 'macos'
+        run: |
+          brew install \
+            coreutils \
+            doctest \
+            ffmpeg \
+            libunistring \
+            ncurses \
+            pandoc \
+            readline
+
+      - name: Install tools and libraries via MSYS2
```

Comment:
It's great to have a method for MSYS2 but this is not the same as native Windows compilation. Ideally this project can be used in Cygwin using these build steps here but also want something more native that terminals like ConEmu (which is a Windows only terminal) can easily use.

Reply:
It would be great if you created another build process under this one for native Windows compilation. It should use "windows-latest" for the platform. I think GH does some stuff behind the scenes with that defined to understand the Windows SDK is needed.

Reply:
> It's great to have a method for MSYS2 but this is not the same as native Windows compilation

I don't think I understand what you mean, though there are a lot of things about compilers, etc. I don't understand because I got a pretty late start with those things after so many years with JS and whatnot.

I am currently using an MSYS2 / MINGW-w64 approach to build executables on Windows 10; having passed `-static` to the compiler I am able to copy the standalone executables to Windows 7 x64 or Windows 10 x64 machines that have nothing at all installed related to MSYS2, MINGW, nor Cygwin. And my executables run just fine in `cmd.exe`. Depending on what the executable does, it can be better (or practically necessary, e.g. if it's printing colored output) to run it in ConEmu or Windows Terminal (running cmd.exe) instead of firing up `cmd.exe` directly.

So, it's not native compilation in the sense that it doesn't use the MS compiler toolchain, but the end result, albeit involving a `-static` compile, is producing a native executable that doesn't depend on anything that doesn't ship with Windows.

I've dabbled with other ways of cross-platform building (for Linux, macOS, and Windows) when it comes to Windows, and I get the most "bang" for my time on Windows by opting for an MSYS2 build context and `-static` compilation.

That being said, I'll repeat that maybe I didn't understand your point. Also, I realize not everyone wants to use MSYS2 for the build context, so perhaps more measures are needed than just getting it to build that way, i.e. this repo's codebase and `CMakeLists.txt` could support building with the MS tooolchain, and other ways too. When it comes to that, I'll be happy to learn and help as I can, but I live in Emacs and bash shells; I've never used Visual Studio and am not really sure where to start when it comes to stuff that's not Unix-like in the way that MSYS2 is. Hopefully other community members can contribute to making this a thing on Windows in multiple best ways possible.

> It should use "windows-latest" for the platform.

Fwiw, the workflow I put together is already using "windows-latest": https://github.com/michaelsbradleyjr/notcurses/blob/github-actions-workflow/.github/workflows/tests.yml#L35